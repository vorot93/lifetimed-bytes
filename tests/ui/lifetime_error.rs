use lifetimed_bytes::Bytes;

fn main() {
    let v = b"hello".to_vec();
    let b = Bytes::from(v.as_slice());

    drop(v);

    println!("oops: {:?}", b);
}
