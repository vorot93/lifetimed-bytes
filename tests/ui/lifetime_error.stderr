error[E0505]: cannot move out of `v` because it is borrowed
 --> $DIR/lifetime_error.rs:7:10
  |
5 |     let b = Bytes::from(v.as_slice());
  |                         - borrow of `v` occurs here
6 |
7 |     drop(v);
  |          ^ move out of `v` occurs here
8 |
9 |     println!("oops: {:?}", b);
  |                            - borrow later used here
